### PR TITLE
update tinyxml to version 10

### PR DESCRIPTION
**Pull request by rettinghaus**
Small update to tinyxml, see https://github.com/leethomason/tinyxml2/releases/tag/10.0.0

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
This seems to introduce a new `PEDANTIC_WHITESPACE` option, which might be useful to solve #19189. Would be interesting the investigate that in the future!

File: src/framework/global/thirdparty/tinyxml/tinyxml2.cpp [hunk 1]
```diff
@@ -23,9 +23,7 @@ distribution.
 
 #include "tinyxml2.h"
 
-#include "mu_patch.h"
```
Comment (cbjeukendrup) [inline src/framework/global/thirdparty/tinyxml/tinyxml2.cpp:26]: This seems to indicate that we patched it at some point. See https://github.com/musescore/MuseScore/commit/d6a586432e2cbd7b3ce03692e3307119c1e3ea5d. We should check if that patch is still needed.

File: src/framework/global/thirdparty/tinyxml/tinyxml2.cpp
Comment (cbjeukendrup) [inline src/framework/global/thirdparty/tinyxml/tinyxml2.cpp:1]: The `100644 → 100755` indicates that the file was made executable, which is unexpected. Better undo that. (Same for other file.)

Comment (Jojo-Schmitz) [reply src/framework/global/thirdparty/tinyxml/tinyxml2.cpp:1]: can you remind me of the git syntax for that?

Comment (rettinghaus) [reply src/framework/global/thirdparty/tinyxml/tinyxml2.cpp:1]: fixed

File: src/framework/global/thirdparty/tinyxml/tinyxml2.cpp (hunk 1 above)
Comment (rettinghaus) [reply src/framework/global/thirdparty/tinyxml/tinyxml2.cpp:26]: I think this is the reason for the failing tests. I added the line again.

Comment (cbjeukendrup) [reply src/framework/global/thirdparty/tinyxml/tinyxml2.cpp:26]: But note that it's not only that line, that's why I mentioned https://github.com/musescore/MuseScore/commit/d6a586432e2cbd7b3ce03692e3307119c1e3ea5d

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**